modules/conduit/include/hephaestus/conduit/zenoh_nodes.h

```diff
@@ -0,0 +1,62 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <cstddef>
+#include <string_view>
+#include <utility>
+
+#include "hephaestus/conduit/node.h"
+#include "hephaestus/conduit/queued_input.h"
+#include "hephaestus/ipc/topic.h"
+#include "hephaestus/ipc/zenoh/publisher.h"
+#include "hephaestus/ipc/zenoh/session.h"
+#include "hephaestus/ipc/zenoh/subscriber.h"
+#include "hephaestus/utils/string/string_literal.h"
+
+namespace heph::conduit {
+
+template <typename InputT>
+class ZenohSubscriberNode {
+public:
+  ZenohSubscriberNode(ipc::zenoh::SessionPtr session, ipc::TopicConfig topic_config, InputT& input)
+    : input_(&input)
+    , subscriber_(std::move(session), std::move(topic_config),
+                  [this](const auto&, const auto& msg) { input_->setValue(*msg); }) {
```

Comment:
`setValue` might return `InputState::OVERFLOW`. I guess you need to do something like:
```
auto backoff = std::chrono::microseconds(1);
while (input->setValue(*msg) != InputState::OK) {
  std::this_thread::sleep_for(backoff);
  backoff = backoff * 2;
}
```